packages/bot-skeleton/src/scratch/blocks/Advanced/Functions/procedures_ifreturn.js

```diff
@@ -123,12 +123,22 @@ Blockly.Blocks.procedures_ifreturn = {
     FUNCTION_TYPES: ['procedures_defnoreturn', 'procedures_defreturn'],
 };
 
-Blockly.JavaScript.procedures_ifreturn = block => {
-    const condition = Blockly.JavaScript.valueToCode(block, 'CONDITION', Blockly.JavaScript.ORDER_NONE) || 'false';
+Blockly.JavaScript.javascriptGenerator.forBlock.procedures_ifreturn = block => {
+    const condition =
+        Blockly.JavaScript.javascriptGenerator.valueToCode(
+            block,
+            'CONDITION',
+            Blockly.JavaScript.javascriptGenerator.ORDER_NONE
+        ) || 'false';
```

Comment:
```suggestion
        ) || false;
```
can u check if it could be an actual boolean instead of string? because the string `'false'` is evaluated as true in javascript

Reply:
This is the code generation part and all the code generated from blockly needs to be in string that is later passed to the interpreter.